File: src/structs.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, val, next: 'ListNode' = None):
        self.val = val
        self.next = next

    def __pow__(self, other: 'ListNode'):
        self.next = other
        return self

    def __str__(self):
        return f"{self.val} -> {self.next}"

    def __eq__(self, other):
        if isinstance(other, self.__class__):
            return self.val == other.val and self.next == other.next
        else:
            return False


# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val, left: 'TreeNode' = None, right: 'TreeNode' = None):
        self.val = val
        self.left = left
        self.right = right

    def __eq__(self, other):
        if isinstance(other, self.__class__):
            return self.val == other.val and self.left == other.left and self.right == other.right
        else:
            return False

File: src/structs.rb
# frozen_string_literal: true

# Definition for singly-linked list.
class ListNode
  attr_accessor :val, :next
  def initialize(val)
    @val = val
    @next = nil
  end

  def **(other)
    @next = other

    self
  end

  def to_s
    "#{@val} -> #{@next || 'nil'}"
  end
end

# Definition for a binary tree node.
class TreeNode
  attr_accessor :val, :left, :right
  def initialize(val)
    @val = val
    @left = nil
    @right = nil
  end
end
